### PR TITLE
feat(camunda-platform): add in & out mapping props

### LINKED ISSUE #89: Migrate variables mapping group
We are going to implement

* two list groups for in / out mappings
- [ ] Type (change to old implementation => no "all" type, this lands in an own section)
- [ ] Source (type === source)
- [ ] Source expression (type === sourceExpression)
- [ ] Local (default = false)

* two groups for in / out mapping propagations
- [ ] Propagate all check box
- [ ] Local (default = false)

image

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #89

We have a change here for the existing in & out mapping implementation: The `all` type is gone from the list items. Instead, there are new propagation groups that handle these cases. 

Note that the wording of the groups is a proposal from my side to better distinguish the to-be-implemented Inputs & Outputs groups.

> In mappings

image

> In mapping propagation

image

Note that the configuration of "local" for the "all mapping" is only possible if the first checkbox is toggled (otherwise the mapping does not exist). The same goes for out mapping propagation.

image

> Out mappings

image

> Out mapping propagation

image

### CODE REVIEW

**Review by andreasgeier — APPROVED** [review]
✅ &nbsp; LGTM

Comment: I know that the order mapping first, then propagation is in line what we did with Zeebe output propagation. But giving that a second thought I was wondering if we should change the order since the propagation flag is a global setting and mappings are more specific. Moreover, in Zeebe the propagation affects the assignment (above) which is unusual. The expected behavior would be to go from top to down and affect properties below, not above.

Therefore, I would suggest that you reconsider it and when it makes sense from a technical perspective, to change the order like that:
- In mapping propagation
- In mappings
- Input
- Out mapping propagation
- Out mappings
- Output

**Comment by pinussilvestrus** [comment]
@andreasgeier thanks for mentioning the ordering of the input/output groups 👍  

I've opened an issue to start the discussion around it: https://github.com/bpmn-io/bpmn-properties-panel/issues/95

**Comment by philippfromme** [comment]
I've found an issue that I suspect might only affect Windows.

w8zKMe6kUF

It seems to be the same issue as https://github.com/camunda/camunda-modeler/issues/2364 which only affects Windows. The issue seems to be about the available space on the screen. As shown above, disabling `height: 100%` on the `bio-properties-panel-container` fixes the issue.

I will create a separate issue for this and merge this pull request regardless.

File: src/provider/camunda-platform/properties/InOutMapping.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  const commandStack = useService('commandStack');
+  const translate = useService('translate');
+
+  const getValue = (mapping) => {
+    return getInOutType(mapping);
+  };
+
+  const setValue = (value) => {
+    let props = {
+      ...DEFAULT_PROPS,
```
Comment (philippfromme) [inline src/provider/camunda-platform/properties/InOutMapping.js:78]: Making sure ony one of `source` and `sourceExpression` is set is a classic use case for a behavior. We will tackle this in the near future. 😉

**Pull request merged**